agent/consul/rpc.go

```diff
@@ -746,6 +745,10 @@ CHECK_LEADER:
 			return true, nil
 		}
 	}
+	// defer to caller to retry
+	if rpcErr != rate.ErrRetryElsewhere {
```

Comment:
tip: use `errors.Is` to handle cases where the error has been [wrapped](https://rollbar.com/blog/golang-wrap-and-unwrap-error).
```suggestion
	if !errors.Is(rpcErr, rate.ErrRetryElsewhere) {
```